### PR TITLE
test floating point casting better

### PR DESCRIPTION
**Pull request by RalfJung**

### CODE REVIEW

File: tests/run-pass/floats.rs [hunk 1]
```diff
@@ -1,37 +1,84 @@
...
+    assert_eq!(std::f32::INFINITY as u32, u32::MAX);
+    assert_eq!(std::f32::MIN as i32, i32::MIN);
+    assert_eq!(std::f32::NEG_INFINITY as i32, i32::MIN);
+    assert_eq!(std::f32::MIN as u32, 0);
+    assert_eq!(std::f32::NEG_INFINITY as u32, 0);
+    assert_eq!(std::f32::NAN as i32, 0);
+    assert_eq!(std::f32::NAN as u32, 0);
+    assert_eq!(u128::MAX as f32, std::f32::INFINITY);
+    assert_eq!((u32::MAX-127) as f32 as u32, u32::MAX); // rounding loss
+    assert_eq!((u32::MAX-128) as f32 as u32, u32::MAX-255); // rounding loss
```
Comment (RalfJung) [inline tests/run-pass/floats.rs]: @hannah-kruppe do these two (and the two "rounding loss" tests below) look like correct behavior to you? I experimentally determined these to be the closest to `u32::MAX` where the output of the roundtrip cast changes.

Comment (RalfJung) [inline tests/run-pass/floats.rs]: @hanna-kruppe do these two (and the two "rounding loss" tests below) look like correct behavior to you? I experimentally determined these to be the closest to `u32::MAX` where the output of the roundtrip cast changes.

Comment (RalfJung) [reply tests/run-pass/floats.rs]: @jodysankey you also seem to know floating point encoding stuff quite well, would be nice if you could have a look.

Comment (hanna-kruppe) [reply tests/run-pass/floats.rs]: Yes, they look exactly right to me. These are transition points from "odd significant + slightly less than 0.5 ULP" (which gets rounded down to uint::MAX - 1 ULP) to "odd significant + 0.5 ULP" (which is rounded up, so saturation kicks in when casting back to integer).

Comment (RalfJung) [reply tests/run-pass/floats.rs]: @hanna-kruppe I only understood half of that, but that half sounds good. ;) Thanks!

**Pull request merged**